app/controllers/users/registrations_controller.rb

```diff
@@ -1,22 +1,22 @@
 # frozen_string_literal: true
-
 class Users::RegistrationsController < Devise::RegistrationsController
-
   include RackSessionsFix
   respond_to :json
 
-  private
-
-  def respond_with(current_user, _opts = {})
-    if resource.persisted?
-      render json: {
-        status: {code: 200, message: 'Signed up successfully.'},
-        data: UserSerializer.new(current_user).serializable_hash[:data][:attributes]
-      }
+  def create
+    contract = Contracts::Users::UserRegistrationContract.new
+    operation = contract.call(params.permit(user: {}).to_h)
+    if operation.success?
+      user = User.new(operation[:user].to_h)
+      if user.save
+        snippets = user.snippets
+        user_resource = UserResource.new(user, snippets)
+        render json: user_resource.custom_json, status: :created
```

Comment:
`new` does not save the user resource.  You need `create` for that.  It looks to me as if you are creating a new user object on each run, but perhaps I'm missing something.